src/handlers/expiry/providers/DateProvider.ts

```diff
@@ -183,11 +183,20 @@ export class DateProvider {
   public static addOneYearMinusOneDayISOString(
     inputDate: Date | string
   ): string {
-    return moment(inputDate)
-      .add(1, "year")
-      .subtract(1, "day")
-      .startOf("day")
-      .toISOString();
+
+    const generateExpiryDate = (day: number = 1): string => {
+      return moment(inputDate)
+          .add(1, "year")
+          .subtract(day, "day")
+          .startOf("day")
+          .toISOString();
+    }
+
+    const isLeapYear = moment(inputDate).isLeapYear();
+
+    return (isLeapYear && moment(inputDate).month() === 1 && moment(inputDate).date() === 29)
```

Comment:
You could wrap the 2nd and 3rd clause here in a function that has a much better readable name e.g. isFebuary29

This would avoid the confusion for people not knowing moment months are 0 indexed and the general readability.

Reply:
This is a good idea, I will move that logic over to a separate function.

Reply:
👍